### PR TITLE
[Go SDK]: Retrieve file size in CreateInitialRestriction in textio.Read

### LINKED ISSUE #25532: [Task][Go SDK]: Use context param and error return value in textio.Read SDF
### What needs to happen?

There is a [TODO](https://github.com/apache/beam/blob/20c8d3a2ba11f3965022e2ba5e12a241a4ec3442/sdks/go/pkg/beam/io/textio/textio.go#L112) in `textio.Read` to retrieve a file's size in `readFn.CreateInitialRestriction` instead of in `sizeFn` once SDF methods support a context.Context parameter and an error return value, which has been added as of #25437.

### Issue Priority

Priority: 2 (default / most normal work should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [X] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by johannaojeling**
Fixes #25532 and refactors `textio.Read` to retrieve the file's size in `readFn.CreateInitialRestriction` instead of in a separate DoFn.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by lostluck** [comment]
Run Go PostCommit

File: sdks/go/pkg/beam/io/textio/textio.go [hunk 1]
```diff
@@ -108,36 +106,30 @@ func expandFn(ctx context.Context, glob string, emit func(string)) error {
 	return nil
 }
 
-// sizeFn pairs a filename with the size of that file in bytes.
-// TODO(https://github.com/apache/beam/issues/20607): Once CreateInitialRestriction supports Context params and
-// error return values, this can be done in readSdfFn.CreateInitialRestriction.
-func sizeFn(ctx context.Context, filename string) (string, int64, error) {
+// readFn reads individual lines from a text file, given a filename and a
+// size in bytes for that file. Implemented as an SDF to allow splitting
```
Comment (lostluck) [inline sdks/go/pkg/beam/io/textio/textio.go:110]: Technically it no longer relies on an input size in bytes. We can to remove that part of the comment now.

**Review by lostluck — APPROVED** [review]
LGTM other than the comment update!

Thank you very kindly!

File: sdks/go/pkg/beam/io/textio/textio.go (hunk 1 above)
Comment (johannaojeling) [reply sdks/go/pkg/beam/io/textio/textio.go]: Good catch, thanks!

**Pull request merged**